common/data_refinery_common/models/models.py

```diff
@@ -939,16 +939,34 @@ def get_synced_file_path(self, force=False, path=None):
 
     @property
     def s3_url(self):
-        """ Render the resulting S3 URL """
+        """ Render the resulting HTTPS URL for the S3 object."""
         return self.get_s3_url()
 
     def get_s3_url(self):
-        """ Render the resulting S3 URL """
+        """ Render the resulting HTTPS URL for the S3 object."""
         if (self.s3_key) and (self.s3_bucket):
             return "https://s3.amazonaws.com/" + self.s3_bucket + "/" + self.s3_key
         else:
             return None
 
+    @property
+    def download_url(self):
+        """ A temporary URL from which the file can be downloaded. """
+        return self.create_download_url()
+
+    def create_download_url(self):
+        """ Create a temporary URL from which the file can be downloaded."""
+        if settings.RUNNING_IN_CLOUD and self.s3_bucket and self.s3_key:
+            return s3.generate_presigned_url(
+                ClientMethod='get_object',
+                Params={
+                    'Bucket': self.s3_bucket,
+                    'Key': self.s3_key
```

Comment:
What is ExpireIn going to be if we don't set it, and do we want the default?

Reply:
OK, looks like I figured this guy out finally. I went with a day because that's how long datasets last. I have a link that shouldn't work this time tomorrow. I'll test that tomorrow but when I switched `ExpiresIn` to a parameter instead of an entry in the `Params` parameter it complained because I used milliseconds and that was more than a week worth of seconds (which is the limit).  Anyway I took that to mean I was finally setting that correctly.